src/inmanta/data/__init__.py

```diff
@@ -5342,19 +5372,36 @@ async def update_fields(self, connection: Optional[asyncpg.connection.Connection
     def to_dict(self) -> Dict[str, Any]:
         self.make_hash()
         dct = super(Resource, self).to_dict()
+
+        # Patch up id, resource_version_id, requires and provides for backward compat with older internal representation
+        # in the DB, it is all resource_id, externally resource_version_id
+        version = self.model
+        rvid = Id.set_version_in_id(self.resource_id, version)
+        dct["resource_version_id"] = rvid
         dct["id"] = dct["resource_version_id"]
+        if "requires" in dct["attributes"]:
+            dct["attributes"]["requires"] = [Id.set_version_in_id(id, version) for id in dct["attributes"]["requires"]]
```

Comment:
I hadn't considered this before: could be slightly confusing (because it's not apparent in the typing of this field) that we have resource ids in the database and version ids in the DTO objects. I'm not sure if we can avoid it but we'll have to be very careful with this.